### PR TITLE
[PD 2933] export ui functionality

### PR DESCRIPTION
**Pull request by mwarnsleyDE**
Merge.
foobar

### CODE REVIEW

File: gulp/src/analytics/components/Header/ExportOptions.js [hunk 1]
```diff
@@ -27,34 +35,47 @@ class ExportOptions extends Component {
   mouseUpHandler() {
     this.mouseIsDownOnExport = false;
   }
+  storeFormatBoxes() {
+    console.log('YOU MIGHT NEED THIS FUNCTION');
```
Comment (darrint) [inline gulp/src/analytics/components/Header/ExportOptions.js]: Are we passing eslint?

File: myreports/helpers.py [hunk 2]
```diff
@@ -4,12 +4,74 @@
...
+
+    :param date: The datetime object.
+    :return: A string representation of the datetime object for use in
+             a report.
+    """
+
+    return date.strftime("%m/%d/%Y")
+
+
+def make_downloadable_report(report, report_presentation, order_by=None,
```
Comment (darrint) [inline myreports/helpers.py]: This looks nice. It could use a docstring.

Comment (almathew) [inline myreports/helpers.py]: I just added one!

File: myreports/datasources/job_views.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        super(JobViewsDataSource, self).__init__()
+
+    def filter_type(self):
+        return JobViewsDataSourceFilter
+
+    def run(self, data_type, company, filter_spec, values_spec):
+        return dispatch_run_by_data_type(
+            self, data_type, company, filter_spec, values_spec)
+
+    def run_unaggregated(self, company, filter_spec, values):
```
Comment (darrint) [inline myreports/datasources/job_views.py]: We don't actually call this do we?

Comment (almathew) [inline myreports/datasources/job_views.py]: No, but I accidentally wrote it and wasn't sure whether or not it would be useful in the future.

Comment (almathew) [inline myreports/datasources/job_views.py]: Same as `help` for the record

File: gulp/src/analytics/components/Header/ExportOptions.js [hunk 4]
```diff
@@ -56,32 +61,20 @@ class ExportOptions extends Component {
...
...NvW3Y3koAfAW5eU70mTIrNZbNPrtplNz5B6E0W84XrD/ViUAEAWA4+ZWbyIHaIyihHYB8KrpWrk8zmAGwsYThbqG66g5lquktUYrAYAfAPhNic4doevv90tTR+XiqEulXfA6VAD4gpC4UF+s82CV1X298QtVOyJ5kFT/RyEA8AdAjkF/v2Z/jlzs6+rWxow2BcD7dg/KGqbeVc1nMIe7dn01wavYsrHSniv6+MDI2TGG7RcAoIjfEG1YdsL88a2jb9ya1sYHfLuDwcUBSuWuHpcoG0zeUO1YwVy6j7fNwGZrewybTz8YouYAlm4DboPuIABQDNYO9HC36pc/i02HjLUDRloAOCLHA0VG8oaQOe6v2kVlWbxUf3y2HA7m8wo2JuliEqsP55mOTfiOCHLpwW+99pm0cxgAAKmI3gyRhtNkgzqZJ1mY1cVm4nYFD9aMKRkd0IcaXB0035gvlzzadok80HOtXNOg1Q8bjy4eJQ80AQCfjNIOrQ5UMlpXZFPrn8Oq2rBtoAB4RxoX78uyjDqqG3JRrW8/dxe1xxc0phtWIdMWTdSdy4EjAGARTzcaNfOKu2i+pqw5+mCIlgeHAgHAG5vn3/E4v78SY0lzlmqod7DP+sij5o5WvdWqCScXSlVzC7TzHACA0qDi+XxcJ3OL32pPI+4q5q6fJrfGtQM2PLGEp5P5+xxr2YEc3WSTlVQi8DGO6HG/fmLFOGnmqiny/EKjUC7PJOJSJ6UmXp5SZtXYXEnPxkZX7cnQDTcDAMwKBgAAAAAAAAAAAAAAAAAAACACAOD+O8fyoaGm8whcDQAUxBpBAAAAOA2AOhhoW01OBiAbBtrWQicD8LlmPSMocPV0LABKmkcpC1nBzMDEL05yIPsGhiUASrrfIg1Tdj3zQH7FSwB+EOjGkWELAGRfAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACOvNMt4h9eZV3Hn/RAAQOcY/brBdHvv2LgCIDADSzcb17ZQEAMA5AFyz8KgHAHC3+V39jPB9CwCwfTwAAAAAAAAAAAAAAAAAAAAAAAAAhAiAFlIc1KFK7kgAoPAXAAAAAAAA2ACgAZnoaPWyC0AKMtGx+o/UzS4AXUgXkZmOVK7tCSEKBN25z0+6jkx1xAeex5QuYic7APwPpAylJWPV+MIAAAAASUVORK5CYII='} alt="Excel"/>
```
Comment (darrint) [inline gulp/src/analytics/components/Header/ExportOptions.js]: I guess we can let it pass for now but these really need to be in `static/`.

File: myreports/tests/test_job_views_datasource.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        data = self.data_source.run_aggregated(self.company, job_views_filter,
+                                               ['state'])
+        self.assertEqual(expected_data, data)
+
+    def test_run_aggregated_filtered_isnull(self):
+        """
+        JobViews with fields set to None should show up in the results
+        with the string "None".
+        """
+        JobVie
```
Comment (darrint) [inline myreports/tests/test_job_views_datasource.py]: This probably doesn't work.

Comment (almathew) [inline myreports/tests/test_job_views_datasource.py]: The appropriate test has been added.

**Comment by almathew** [comment]
This should be functionally complete and ready for review.

Tests will be passing as soon as MemSQL is available on Jenkins.

**Comment by mwarnsleyDE** [comment]
Tested and the downloading of CSV and Excel files works. I currently don't have any data locally so someone else might want to take a stab at it using some kind of data and see how it's currently working for them.

**Pull request merged**